File: Programmers/Heap/더 맵게.py
import heapq as hq

def solution(scoville, K):
    answer = 0
    hq.heapify(scoville)
    
    while True:
        b1 = hq.heappop(scoville)
        if b1 > K:
            return answer
        else:
            try:
                b2 = hq.heappop(scoville)
                hq.heappush(scoville,b1+b2*2)
                answer += 1
            except:
                return -1    
    return answer

File: Programmers/Heap/더 맵게.cpp
#include <string>
#include <vector>
#include <queue>

using namespace std;

int solution(vector<int> scoville, int K) {
    int answer = 0;
    priority_queue<int> heap;
    for(int i=0 ;i<scoville.size();++i)
        heap.push(-scoville[i]);
    
    int top1, top2;
    while(-heap.top() < K ){
        top1 = heap.top();
        heap.pop();
        if(heap.size()==0)
            return -1;
        top2 = heap.top();
        heap.pop();
        heap.push(top1+top2*2);
        answer++;
    }
    
    return answer;
}